Bottom: 48788dbfcbcc31919946d94778b4c5c682c4c911
Top:    0057e32f2d79234c1c58935e617fcc62008293c3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-27 17:55:35 +0100

Refresh of use-trackball-to-allow-rotatio

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 5fa78ea..0370ffc 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1548,14 +1548,14 @@ ghid_event_to_pcb_coords (int event_x, int event_y, Coord *pcb_x, Coord *pcb_y)
 bool
 ghid_pcb_to_event_coords (Coord pcb_x, Coord pcb_y, int *event_x, int *event_y)
 {
-  *event_x = view_matrix[0][0] * pcb_x +
-             view_matrix[0][1] * pcb_y +
-             view_matrix[0][2] * 0 +
-             view_matrix[0][3] * 1;
-  *event_y = view_matrix[1][0] * pcb_x +
-             view_matrix[1][1] * pcb_y +
-             view_matrix[1][2] * 0 +
-             view_matrix[1][3] * 1;
+  *event_x = view_matrix[0][0] * (float)pcb_x +
+             view_matrix[0][1] * (float)pcb_y +
+             view_matrix[0][2] * 0. +
+             view_matrix[0][3] * 1.;
+  *event_y = view_matrix[1][0] * (float)pcb_x +
+             view_matrix[1][1] * (float)pcb_y +
+             view_matrix[1][2] * 0. +
+             view_matrix[1][3] * 1.;
 
   return true;
 }
